fix: Fix async execution demo

NeilFraser opened the pull request:
It was completely broken during the move from blockly-demos to github.io.

Also update to latest version of JS-Interpreter (performance improvements).

Commit:
Fix async execution demo

It was completely broken during the move from blockly-demos to github.io.

Also update to latest version of JS-Interpreter (performance improvements).

NeilFraser commented on examples/interpreter-demo/async-execution.html:
```diff
@@ -43,166 +42,140 @@ <h1>Asynchronous Execution with JS-Interpreter</h1>
-
-
-    // Exit is used to signal the end of a script.
-    Blockly.JavaScript.addReservedWords('exit');
```
I have absolutely no idea what 'exit' is about.  AnmAtAnm added it in 2018 for unknown reasons.

BeksOmega commented on examples/interpreter-demo/async-execution.html:
```diff
@@ -43,166 +42,140 @@ <h1>Asynchronous Execution with JS-Interpreter</h1>
     var demoWorkspace = Blockly.inject('blocklyDiv',
         {media: 'https://unpkg.com/blockly/media/',
-         toolbox: toolboxJson});
+         toolbox: document.getElementById('toolbox')});
```
Why did you switch it back to using XML rather than JSON? Was this an accidental reversion?

NeilFraser replied:
The XML toolbox contains the 'wait' block this demo needs.  The JSON toolbox (used by the other demos) does not have this block.  Not having the 'wait' block renders this demo useless.

BeksOmega replied:
Can you add the 'wait' block to the JSON toolbox?

NeilFraser replied:
What's the advantage of using JSON?  The JSON toolbox looks significantly larger.  And our developer tools only generates XML.

BeksOmega replied:
We are trying to standardize to using JSON toolboxes, because (1) the XML serialization system is iceboxed, (2) the XML toolbox does not work well within certain frameworks (e.g. React), and (3) consistency with all of our other demos.

BeksOmega commented on examples/interpreter-demo/serialize.js:
```diff
@@ -130,6 +130,12 @@ function deserialize(json, interpreter) {
       delete obj.end;
       var locText = jsonObj['loc'];
       if (locText) {
+        // Turn a serialized string '1:0-4:21 code' into a location object:
```
I think it makes sense to break this out into a subprocedure.

NeilFraser replied:
Done.

BeksOmega commented on examples/interpreter-demo/serialize.js:
```diff
@@ -344,7 +350,13 @@ function serialize(interpreter) {
       var name = names[j];
       if (jsonObj['type'] === 'Node' && name === 'loc') {
-        // Compactly serialize the location objects on a Node.
+        // Compactly serialize the location objects on a Node:
```
Same as above.

NeilFraser replied:
Done.

BeksOmega commented on examples/interpreter-demo/step-execution.html:
```diff
@@ -28,16 +28,17 @@ <h1>Step Execution with JS-Interpreter</h1>
   will be executed on the next step.</p>
 
-  <p>&rarr; <a href="https://developers.google.com/blockly/guides/configure-blockly/web/running-javascript#js_interpreter">More info on running code with JS-Interpreter</a></p>
+  <p>&rarr; <a href="https://developers.google.com/blockly/guides/app-integration/running-javascript#js-interpreter">More info on running code with JS-Interpreter</a></p>
```
Yay for updating docs links!

Commit:
Factor out location serialization/deserialization

Mirrors this commit on JS-Interpreter: https://github.com/NeilFraser/JS-Interpreter/commit/ea51a5045db034909fafc82a7edf48ff3041228e

Commit:
Port today's changes from JS-Interpreter.

Commit:
Remove ES3 support from Acorn.

Commit:
Interpreter code health updates

Commit:
Convert XML to JSON

BeksOmega commented on examples/interpreter-demo/async-execution.html:
```diff
@@ -42,140 +42,294 @@ <h1>Asynchronous Execution with JS-Interpreter</h1>
-  </xml>
+  <script>
+    var toolboxJson = {
+  "contents": [
```
Can you fix the indentation here?

NeilFraser replied:
Done.

BeksOmega reviewed (approved):
Woo! thanks for converting the toolbox back to JSON :D LGTM!

Commit:
var to let/const
